### PR TITLE
Add description forrunner_registration_token and replace underscore with dash

### LINKED ISSUE #87: Getting a error when runner_registration_token contains an underscore.
When the `aws_ssm_parameter.runner_registration_token` contains an underscore I'm getting the error:
`Error creating machine: Invalid hostname specified. Allowed hostname chars are: 0-9a-zA-Z . -`.
I solved it by recreating a runner to get a token without underscore. But maybe it makes sense to substitute underscore with a dash?

### PR DESCRIPTION
**Pull request by midnightrun**
Prevent tokens to contain underscore

## Description
Fixes #87 

## Migrations required
No

## Verification
Run code without any problems. Tested within `terraform console`.

## Documentation
Done.

### CODE REVIEW

**Comment by midnightrun** [comment]
I believe using a regex would be even nicer but just wanted to prevent this behaviour just somehow. 

Thoughts?

**Comment by npalm** [comment]
@midnightrun thanks will have a look at the PR next days

File: main.tf [hunk 1]
```diff
@@ -142,7 +142,7 @@ data "template_file" "gitlab_runner" {
     runners_token                           = var.runners_token
     secure_parameter_store_runner_token_key = local.secure_parameter_store_runner_token_key
     secure_parameter_store_region           = var.aws_region
-    gitlab_runner_registration_token        = var.gitlab_runner_registration_config["registration_token"]
```
Comment (npalm) [inline main.tf]: Is this really working? In this way the registration token is changed. It also seems the bug is solved in the gitlab runner: 
https://gitlab.com/gitlab-org/gitlab-runner/merge_requests/1144#9c5c461492d69bfd6a2ce40727e0b81b49a19e5b

Comment (midnightrun) [reply main.tf]: Ah, sorry. I just tested if the substitution is working. If Gitlab really fixed this. This PR is obsolete and can be closed. Thanks!

Comment (npalm) [reply main.tf]: I am going to test is today, just got a token with an underscore.

Comment (npalm) [reply main.tf]: I have just created a runner with a token containing an `_` the runner is created without an error and working as normal. Seems the issue is resolved by GitLab.

**Comment by npalm** [comment]
@midnightrun Thanks proposing a solution for #87 but the issue is resolved by GitLab since 11.7 (https://gitlab.com/gitlab-org/gitlab-runner/merge_requests/1144#9c5c461492d69bfd6a2ce40727e0b81b49a19e5b) So I will close you PR without merging.

**Pull request closed without merging**